packages/core/src/new-api/internal/reconciliation/utils.ts

```diff
@@ -68,7 +67,7 @@ export function resolveModuleParameter(
   return moduleParamValue;
 }
 
-export function safeToString(potential: ModuleParameterType | undefined) {
+export function safeToString(potential: unknown) {
```

Comment:
TBH I don't like this api. What does `safe` mean? When can I use this? What should I expect to get? The `unknown` makes it harder to accept, as it now can accept anything, leading to mistakes.

I'd rather have one or more api with a clear semantics, like `moduleArgumentToErrorString`.

Reply:
Agreed. I have added `accountRuntimeValueToErrorString`, `moduleParameterToErrorString` and `addressToErrorString` as a replacement.